library/network/src/modules/SuSEFirewall.rb

```diff
@@ -1617,6 +1617,71 @@ def SetIgnoreLoggingBroadcast(zone = nil, bcast)
       nil
     end
 
+    # Function returns list of allowed ports for zone and protocol
+    #
+    # @param [String] zone
+    # @param [String] protocol
+    # @return	[Array<String>] of allowed ports
+    def GetAllowedServicesForZoneProto(zone, protocol)
+
+      Yast.import "SuSEFirewallServices"
+
+      ports = []
+      protocol.downcase!
```

Comment:
Ruby passes arguments by reference so this will modify the argument:
```console
$ cat param-by-reference.rb
def good?(s)
  s.downcase!
  s == "good"
end

a = "GOOD"
if good?(a)
  puts "#{a} morning"
end
$ ruby param-by-reference.rb
good morning
```
Sometimes it is the desired effect of a method but it is not the case for GetAllowedServicesForZoneProto.